modules/class.jetpack-calypsoify.php

```diff
@@ -183,12 +183,47 @@ public function check_page() {
 	}
 
 	public function is_page_gutenberg() {
+		if ( ! Jetpack_Gutenberg::is_gutenberg_available() ) {
+			return false;
+		}
+
 		$page = wp_basename( esc_url( $_SERVER['REQUEST_URI'] ) );
-		return Jetpack::is_gutenberg_available() && (
-				false !== strpos( $page, 'post.php' ) ||
-				false !== strpos( $page, 'post-new.php' ) ||
-				false !== strpos( $page, 'revision.php' )
-			);
+
+		/**
+		 * Allow third-party plugins to register support for their post types in the full-screen Calypsoify gutenberg mode.
+		 *
+		 * @module calypsoify
+		 *
+		 * @since 6.7.0
+		 *
+		 * @param array $post_types Allowed post types.
+		 */
+		$allowed_gutenberg_post_types = apply_filters( 'jetpack_calypsoify_allowed_gutenberg_post_types', array( 'post', 'page' ) );
```

Comment:
Instead of making our own filter, could we use a Core filter?
5.0 will include a `use_block_editor_for_post_type` filter.
https://github.com/WordPress/WordPress/blob/5.0-branch/wp-admin/includes/post.php#L1923-L1946